### PR TITLE
allow initing and solving with u0=nothing

### PR DESCRIPTION
**Pull request by oscardssmith**
Take 2 of https://github.com/SciML/OrdinaryDiffEq.jl/pull/1818 This time it works with external solvers. @ChrisRackauckas how should I add a test for this? I can't `init` a problem within DiffEqBase since none of the algs are defined.

### CODE REVIEW

**Comment by oscardssmith** [comment]
hmm, the invalidation is the new `solve!` method I added (I think). I'm not sure what the best fix is.

**Comment by ChrisRackauckas** [comment]
The null integrator should probably interpolate to the analytical solution?

File: test/downstream/null_de.jl [hunk 1]
```diff
@@ -13,10 +13,16 @@ sol = solve(prob, Tsit5())
...
-sol = solve!(integ)
-@test sol[x] == [0.0, 0.0]
-@test sol[y] == [0.0, 0.0]
+for kwargs in [
+    Dict(:saveat => 0:0.1:1),
+    Dict(:save_start => false),
+    Dict(:save_end => false),
+]
+    sol = solve(prob, kwargs...)
+    init_sol = solve!(init(prob, kwargs...))
```
Comment (ChrisRackauckas) [inline test/downstream/null_de.jl:22]: and test the iteration?

**Comment by oscardssmith** [comment]
I think this is finally done. In the process I discovered a bug in `build_null_solution` where
```
f(u,p,t) = -u
prob = ODEProblem(nothing, (0,1.0))
solve(prob, tspan=0.1)
```
didn't expand the `0.1` to `0.0:0.1:1.0`

**Comment by ChrisRackauckas** [comment]
`solve(prob, tspan=0.1)` should be a warning about an incorrect kwarg?

**Comment by ChrisRackauckas** [comment]
The test failures look real. I don't think the if statement is only catching the right cases.

**Comment by ChrisRackauckas** [comment]
The test case looks fine if downstream passes.

**Comment by ChrisRackauckas** [comment]
https://github.com/SciML/DiffEqBase.jl/actions/runs/3875552613/jobs/6608254597#step:6:618 way too much is caught in this branch.

**Comment by oscardssmith** [comment]
@ChrisRackauckas I think this is finally ready to go.

**Comment by ChrisRackauckas** [comment]
Except tests are failing.

**Comment by oscardssmith** [comment]
The failing test is from sundials, and happens when constructing `JacVec(Lotka, ones(2))` which never inits or solves anything so it's hard to see how it could possibly related. The failure looks like it's `static_hasmethod` related.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by oscardssmith** [comment]
thanks!
